lib/Dialect/FIRRTL/Transforms/InferWidths.cpp

```diff
@@ -1211,10 +1211,14 @@ LogicalResult InferenceMapping::mapOperation(Operation *op) {
     if (auto mux = dyn_cast<MuxPrimOp>(op))
       if (hasUninferredWidth(mux.getSel().getType()))
         allWidthsKnown = false;
-    if (!hasUninferredWidth(result.getType()))
-      declareVars(result, op->getLoc());
-    else
-      allWidthsKnown = false;
+    // Only consider FIRRTL types for width constraints. Ignore any foreign
+    // types as they don't participate in the width inference process.
+    if (auto resultTy = result.getType().dyn_cast<FIRRTLType>()) {
```

Comment:
This could be re-ordered to do a `continue` at the top of the if the dyn_cast doesn't succeed. This may make the logic somewhat cleaner and avoid one nesting level.

Reply:
Yeah that's an excellent suggestion. Applied.